packages/sanity/src/core/studio/workspaceLoader/WorkspaceRouterProvider.tsx

```diff
@@ -30,15 +33,30 @@ export function WorkspaceRouterProvider({
   const history = useRouterHistory()
   const router = useMemo(() => createRouter({basePath, tools}), [basePath, tools])
   const [state, onNavigate] = useRouterFromWorkspaceHistory(history, router, tools)
+  const {onStudioError} = useSource()
```

Comment:
catching the ErrorBoundary from `StudioErrorBoundary` since it has the context it needs here